components/esp_lvgl_port/esp_lvgl_port_fs.c

```diff
@@ -0,0 +1,259 @@
+{
+    (void) drv;
+
+    *br = fread(buf, 1, btr, (FILE *) file_p);
+
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Write into a file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable
+ * @param buf       pointer to a buffer with the bytes to write
+ * @param btr       Bytes To Write
+ * @param br        the number of real written bytes (Bytes Written). NULL if unused.
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_write(lv_fs_drv_t *drv, void *file_p, const void *buf, uint32_t btw, uint32_t *bw)
+{
+    (void) drv;
+
+    *bw = fwrite(buf, 1, btw, (FILE *) file_p);
+
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Set the read write pointer. Also expand the file size if necessary.
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable. (opened with lv_ufs_open )
+ * @param pos       the new position of read write pointer
+ * @param whence    tells from where to interpret the `pos`. See @lv_fs_whence_t
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_seek (lv_fs_drv_t *drv, void *file_p, uint32_t pos, lv_fs_whence_t whence)
+{
+    (void) drv;
+
+    int ret_val = fseek((FILE *) file_p, pos, whence);
+
+    if (0 != ret_val) {
+        return LV_FS_RES_UNKNOWN;
+    }
+
+    return LV_FS_RES_OK;
+}
+/**
+ * Give the position of the read write pointer
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable.
+ * @param pos_p     pointer to to store the result
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_tell (lv_fs_drv_t *drv, void *file_p, uint32_t *pos_p)
+{
+    (void) drv;
+
+    *pos_p = ftell((FILE *) file_p);
+
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Initialize a 'lv_fs_dir_t' variable for directory reading
+ * @param drv       pointer to a driver where this function belongs
+ * @param path      path to a directory
+ * @return          pointer to the directory read descriptor or NULL on error
+ */
+static void *fs_dir_open (lv_fs_drv_t *drv, const char *path)
+{
+    (void) drv;
+
+    DIR *d = NULL;
+
+    d = opendir(path);
+
+    if (NULL == d) {
+        return LV_FS_RES_UNKNOWN;
+    }
+
+    return d;
+}
+
+/**
+ * Read the next filename form a directory.
+ * The name of the directories will begin with '/'
+ * @param drv       pointer to a driver where this function belongs
+ * @param rddir_p   pointer to an initialized 'lv_fs_dir_t' variable
+ * @param fn        pointer to a buffer to store the filename
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_dir_read (lv_fs_drv_t *drv, void *rddir_p, char *fn)
+{
+    lv_fs_res_t res = LV_FS_RES_NOT_IMP;
+
+    readdir((DIR *) rddir_p);
```

Comment:
I don't understand. It returns not implemented, but I see some function make something...